apps/dh/api-dh/source/DataHub.WebApi/Clients/Wholesale/V3/WholesaleClient.cs

```diff
@@ -664,6 +677,104 @@ public virtual async System.Threading.Tasks.Task<ProcessStepResultDto> GetProces
             }
         }
 
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <summary>
+        /// Downloads a compressed settlement report for the specified parameters.
+        /// </summary>
+        /// <param name="gridAreaCodes">A list of grid areas to create the settlement report for.</param>
+        /// <param name="processType">Currently expects BalanceFixing only.</param>
+        /// <param name="periodStart">The start date and time of the period covered by the settlement report.</param>
+        /// <param name="periodEnd">The end date and time of the period covered by the settlement report.</param>
+        /// <param name="energySupplier">Optional GLN/EIC identifier for an energy supplier.</param>
+        /// <param name="csvFormatLocale">Optional locale used to format the CSV file, e.g. da-DK. Defaults to en-US.</param>
+        /// <exception cref="ApiException">A server side error occurred.</exception>
+        public virtual async System.Threading.Tasks.Task<FileResponse> DownloadAsync(System.Collections.Generic.IEnumerable<string> gridAreaCodes, ProcessType processType, System.DateTimeOffset periodStart, System.DateTimeOffset periodEnd, string energySupplier = null, string csvFormatLocale = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken))
+        {
+            if (gridAreaCodes == null)
```

Comment:
Wouldn't the more concise version: ArgumentNullException.ThrowIfNull look better here?

Reply:
It would. :D